src/backend/executor/nodeDML.c

```diff
@@ -327,6 +322,20 @@ ExecInitDML(DML *node, EState *estate, int eflags)
 		}
 	}
 
+	/*
+	 * If table is replicated, update es_processed only at one segment.
+	 * It allows not to adjust es_processed at QD after all executors send
+	 * the same value of es_processed.
+	 */
+	if (Gp_role == GP_ROLE_EXECUTE)
+	{
+		if (GpPolicyIsReplicated(resultRelInfo->ri_RelationDesc->rd_cdbpolicy) &&
+			GpIdentity.segindex != (gp_session_id % resultRelInfo->ri_RelationDesc->rd_cdbpolicy->numsegments))
```

Comment:
Why do we always re-calculate segindex that updates `es_processed`? Can't we just set it to segment index 0?

Reply:
It was taken from the `Explicit gather motion` node logic: https://github.com/arenadata/gpdb/blob/8f17c3762cd88d448b5c6ec8e24114cf97d5a04c/src/backend/executor/nodeMotion.c#L330